packages/block-library/src/calendar/edit.js

```diff
@@ -61,9 +61,13 @@ class CalendarEdit extends Component {
 }
 
 export default withSelect( ( select ) => {
+	const coreEditorSelect = select( 'core/editor' );
```

Comment:
This is very concerning as it might break many 3rd party blocks when they are inserted. Do you have any issue where you track discussion related to how to ensure that blocks are rendered in the proper context?

Reply:
What if other block explicitly imports core/editor package making this store available?

Reply:
> What if other block explicitly imports core/editor package making this store available?

It may not break the block, but might produce weird results. For instance, there's no post object or something like that. This is one of those things we need to document properly before the release. It's still on the table that the blocks should opt-in or opt-out of the widgets screen. I think we need more wide usage of the widgets screen here before making an informed decision.

Reply:
Yes, some blocks that rely on the editor module may not work on the widget screen, because we don't have a post there. There is not exactly breaking back-compatibility because these blocks were created for the post editor and they still work on the post editor; I guess an option to disable/enable blocks from the widget screen may be a possibility.